app/ui/src/app/platform/types/integration/integration.models.ts

```diff
@@ -23,24 +23,34 @@ export const ERROR = 'Error';
 
 export type IntegrationStatus = 'Pending' | 'Published' | 'Unpublished' | 'Error';
 
-export interface Integration extends BaseEntity {
+export interface IntegrationOverview extends BaseEntity {
+  version?: number;
+  tags: Array<string>;
   description?: string;
+  draft: boolean;
+  deployments?: Array<DeploymentOverview>;
+  currentStatus: IntegrationStatus;
```

Comment:
Isn't it `currentState` on the actual response back?  Just checked here and I see `currentState` and `targetState` in an integration overview object.

Reply:
You're right, I need to rename this again

Reply:
In any event, naming something like this as `currentState` was a very bad idea :( 

Same as moving tokens around mapped to strings, such as `DRAFT` or `PUBLISHED`. We do have enums in TypeScript for us and this forces the application consumer to be aware of each and every token ever created, which simply can't scale.